### PR TITLE
Add docs for `@configu-integrations/my-sql`

### LINKED ISSUE #596: Documentation: `MySQLConfigStore`
### Suggestion

Document the `MySQLConfigStore` integration with all the necessary details that explain what the store is and how to use it.

### Motivation

- Introduces users to the [ConfigStore](https://docs.configu.com/introduction/concepts#configstore).
- Improves usability of the store by providing usage examples.
- Improves the completeness of the integration package.

### Context

- Integration docs: https://dev.mysql.com/doc
- Integration SDK docs: https://typeorm.io/data-source-options#mysql--mariadb-data-source-options
- Example PR of a store doc contribution: #575

### PR DESCRIPTION
**Pull request by abhirajadhikary06**
Closes #596 

@RichardAkman can you please review this PR.

### CODE REVIEW

**Comment by abhirajadhikary06** [comment]
@RichardAkman can you please review this PR.

File: packages/integrations/stores/my-sql/README.md
Comment (RichardAkman) [inline packages/integrations/stores/my-sql/README.md:25]: Align with section with the `configuration` section from the reference PR https://github.com/configu/configu/pull/575.
- Rename the section to `Configuration`.
- Move it above above the `Usage` section.
- Instead of listing all of the TypeORM data source options, let's point to the relevant part of TypeORM docs. TypeORM's docs are more robust since they created the underlying SDK and we can avoid doc drifts.

File: packages/integrations/stores/my-sql/README.md
Comment (RichardAkman) [inline packages/integrations/stores/my-sql/README.md:24]:
```suggestion
```
No need for this section, these dependencies should be included with the package itself. Mentioned in the above comment.

File: packages/integrations/stores/my-sql/README.md
Comment (RichardAkman) [inline packages/integrations/stores/my-sql/README.md:54]: Please align the usage section with the one from the reference PR #575. The users of this store are not expected to use the package directly but through one of our interfaces. Make sure to adjust the configuration options of the `.configu` snippet.

**Review by RichardAkman — CHANGES_REQUESTED** [review]
Superb work 🎖️  We should consolidate the structure of this file with the structure from the reference PR #575 to ensure alignment of all store doc files. Review my comments before we can proceed with merging this PR. 

Some of the extra sections you've added are great, I'll consider if they're relevant or require adjustments while you apply my other comments ⭐

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:8]: Changes done.

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md]: Changes done.

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:25]: Rename to Configuration --> Moved it above Usage section --> listed relevant TypeORM's docs.

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:24]: Okay I'll be removing this also.

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:54]:
```suggestion
## Usage

### `.configu` store declaration

To use the MySQLConfigStore with Configu, declare it in the `.configu` file as shown below:

```yaml
stores:
  my-store:
    type: my-sql
    configuration:
      url: mysql://<username>:<password>@<host>:<port>/<database>
      ssl:
        ca: <path-to-server-cert.pem>
        key: <path-to-client-key.pem>
        cert: <path-to-client-cert.pem>
```

### CLI examples

#### Upsert command

To upsert configuration data into the MySQLConfigStore, use the following CLI command:

```bash
configu upsert --store "my-store" --set "test" --schema "./start.cfgu.json" \
    -c "GREETING=hey" \
    -c "SUBJECT=configu mysql store"
``` 
```

Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:54]: Updated the Usage keep it relevant to #575

File: packages/integrations/stores/my-sql/README.md
Comment (abhirajadhikary06) [reply packages/integrations/stores/my-sql/README.md:14]: ### MySQL Connection Configuration Parameters

| Parameter  | Description                                            | Default     |
|------------|--------------------------------------------------------|-------------|
| `url`      | Connection URL for the MySQL server. Overrides other options. | None        |
| `host`     | Database host.                                          | `localhost` |
| `port`     | Database port.                                          | `3306`      |
| `username` | Username for database authentication.                   | None        |
| `password` | Password for database authentication.                   | None        |
| `database` | Name of the database to connect to.                     | None        |

For additional parameters, refer to the [TypeORM documentation](https://typeorm.io/#/connection-options).

---

Additionally, in the **`.configu` store declaration** section, it shows how to declare credentials in a `.configu` file:

```yaml
stores:
  my-store:
    type: my-sql
    configuration:
      url: mysql://<username>:<password>@<host>:<port>/<database>
      ssl:
        ca: <path-to-server-cert.pem>
        key: <path-to-client-key.pem>
        cert: <path-to-client-cert.pem>
```

**Comment by abhirajadhikary06** [comment]
Hi @RichardAkman,
I hope you’re doing well! I just wanted to check in and see if there are any updates on this. I understand the maintainer may be busy, so no rush—just looking forward to any feedback when it’s convenient.

**Comment by abhirajadhikary06** [comment]
I just wanted to check in to see if any further changes are needed for this PR. I’m eager to know if there are any remaining issues to address, as it hasn’t been merged yet.

**Review by RichardAkman — APPROVED** [review]
LGTM ⭐ 

Following #605, I've made a few adjustments to the documentation to ensure alignment with other TypeORM extending stores.

**Comment by abhirajadhikary06** [comment]
@RichardAkman Thank you for the fast response 😊

**Comment by RichardAkman** [comment]
> @RichardAkman Thank you for the fast response 😊

Absolutely @abhirajadhikary06, thank you for your contribution!

**Comment by pelegpor** [comment]
Hi @abhirajadhikary06 ,

Here are the details for claiming your Hacktoberfest prizes! 🎉 Check out [#465  (comment)](https://github.com/configu/configu/issues/465#issuecomment-2498550259) for more information or simply fill out the form here: [Claim Your Swag](https://share-eu1.hsforms.com/2SVc5VKY3Thi04Bvjh17WKw2e62o0).  

Remember, you only need to fill out the form once for all your PRs. 😊

**Pull request merged**
